File: bab_7_perulangan/perulangan_2_7.c
#include <stdio.h>
#include <string.h>
int main() {
    // Deklarasi dan inisialisasi array dengan data string
    char strings[24][20] = {
        "", "Tahajud", "","","Subuh", "",
        "Duha","","","","","",
        "Dhuhur","","","Asyar","","",
        "Maghrib","Isyak","","","",""
    };

    char tekan_tombol;
    int iterasi_hari = 1;
    do {
        switch (iterasi_hari)
        {
        case 1:
            printf("Jadwal Sholat hari ini\n");
            break;
        case 2:
            printf("Jadwal Sholat besuk\n");
            break;
        case 3:
            printf("Jadwal Sholat lusa\n");
            break;
        default:
            printf("Jadwal Sholat %d hari yang akan datang\n", iterasi_hari);
            break;
        }
        iterasi_hari +=1;
        for (int i = 0; i < 24; i++) {
            if(strcmp(strings[i], "") != 0)
                printf("Jam %d: %s\n", i, strings[i]);
        }

        printf("Lihat jadwal Sholat hari berikutnya?(Y/N): ");
        tekan_tombol = getchar();
        getchar(); // mengabaikan tombol enter
    }while(tekan_tombol != 'N');
    return 0;
}

File: bab_7_perulangan/perulangan_2_7.py
#Simpan dengan nama file perulangan_2_7.py
iterasi_hari = 1
ulangi = True
while ulangi == True:
    if(iterasi_hari == 1):
        print("Jadwal Sholat hari ini")
    elif(iterasi_hari == 2):
        print("Jadwal Sholat untuk besuk")
    elif(iterasi_hari == 3):
        print("Jadwal Sholat untuk lusa")
    else :
        print("Jadwal Sholat untuk %d hari yang akan datang" %iterasi_hari)
    iterasi_hari += 1

    for jam in range(24):
        if(jam == 1):
            print("Jam %d waktunya sholat sunnah Tahajud" %jam)
        elif(jam == 4):
            print("Jam %d waktunya sholat Subuh" %jam)
        elif(jam == 6):
            print("Jam %d waktunya sunnah Dhuha" %jam)
        elif(jam == 12):
            print("Jam %d waktunya sholat DZuhur" %jam)
        elif(jam == 15):
            print("Jam %d waktunya sholat Asyar" %jam)
        elif(jam == 18):
            print("Jam %d waktunya sholat Maghrib" %jam)
        elif(jam == 19):
            print("Jam %d waktunya sholat Isyak" %jam)
            
    tekan_tombol = input("Lihat jadwal Sholat hari berikutnya?(Y/N): ")
    if(tekan_tombol == "N"):
        ulangi = False




